matplotcheck/tests/test_base_data.py

```diff
@@ -72,6 +83,143 @@ def test_assert_xydata_xlabel_fails(pt_bar_plt, pd_df):
     plt.close()
 
 
+def test_assert_xydata_xlabel_text():
+    "Tests the xlabels flag on xydata works to test labels with text data"
+    data = {
+        "months": ["Jan", "Feb", "Mar", "Apr", "May", "June", "July"],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    df = pd.DataFrame(data)
+
+    fig, ax = plt.subplots()
+    df.plot("months", "data", kind="bar", ax=ax)
+    axis = plt.gca()
+
+    pt = PlotTester(axis)
+    pt.assert_xydata(df, xcol="months", ycol="data", xlabels=True)
+
+    plt.close()
+
+
+def test_assert_xydata_xlabel_text_fails():
+    "Tests the xlabels flag on xydata fails when testing labels with wrong text data"
+    correct_data = {
```

Comment:
this dataframe should definitely be a fixture. then you can create a plot tester object from it as a second fixture. similar to below.